make/autoconf/basic_windows.m4

```diff
@@ -34,9 +34,10 @@ AC_DEFUN([BASIC_SETUP_PATHS_WINDOWS],
       OPENJDK_BUILD_OS_ENV=windows.wsl1
     else
       # This test is not guaranteed, but there is no documented way of
-      # distinguishing between WSL1 and WSL2. Assume only WSL2 has WSL_INTEROP
-      # in /run/WSL
-      if test -d "/run/WSL" ; then
+      # distinguishing between WSL1 and WSL2.
+      # Check whether "Hyper-V" appears in /proc/interrupts because WSL2 runs on Hyper-V.
+      grep Hyper-V /proc/interrupts > /dev/null 2>&1
```

Comment:
You could use the -q switch to grep instead of piping to /dev/null.

Reply:
Thanks for your review! I updated  to add `-q` and to use `$GREP` instead of `grep`.